iothub/device/src/IotHubConnection.cs

```diff
@@ -337,27 +339,29 @@ async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri, TimeSpa
             websocket.Options.AddSubProtocol(WebSocketConstants.SubProtocols.Amqpwsb10);
 
             // Check if we're configured to use a proxy server
-            IWebProxy webProxy = WebRequest.DefaultWebProxy;
-            Uri proxyAddress = null;
-#if !NETSTANDARD1_3 && !NETSTANDARD2_0
-            proxyAddress = webProxy != null ? webProxy.GetProxy(websocketUri) : null;
-#endif
-            if (!websocketUri.Equals(proxyAddress))
+            IWebProxy webProxy = this.AmqpTransportSettings.Proxy;
+
+            try
+            {
+                if (webProxy != DefaultWebProxySettings.Instance)
+                {
+                    // Configure proxy server
+                    websocket.Options.Proxy = webProxy;
+                    if (Logging.IsEnabled)
+                        Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
+                }
+            }
+            catch (PlatformNotSupportedException)
             {
-                // Configure proxy server
-                websocket.Options.Proxy = webProxy;
+                // .NET Core 2.0 doesn't support proxy. Ignore this setting.
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
```

Comment:
Writing it this way is dangerous: either add braces or write the statement on the same line.